ui/common/src/commonMain/kotlin/social/androiddev/common/composables/text/MastodonTextField.kt

```diff
@@ -0,0 +1,301 @@
+        if (disabledTextColor != other.disabledTextColor) return false
+        if (cursorColor != other.cursorColor) return false
+        if (errorCursorColor != other.errorCursorColor) return false
+        if (focusedIndicatorColor != other.focusedIndicatorColor) return false
+        if (unfocusedIndicatorColor != other.unfocusedIndicatorColor) return false
+        if (errorIndicatorColor != other.errorIndicatorColor) return false
+        if (disabledIndicatorColor != other.disabledIndicatorColor) return false
+        if (leadingIconColor != other.leadingIconColor) return false
+        if (disabledLeadingIconColor != other.disabledLeadingIconColor) return false
+        if (errorLeadingIconColor != other.errorLeadingIconColor) return false
+        if (trailingIconColor != other.trailingIconColor) return false
+        if (disabledTrailingIconColor != other.disabledTrailingIconColor) return false
+        if (errorTrailingIconColor != other.errorTrailingIconColor) return false
+        if (backgroundColor != other.backgroundColor) return false
+        if (focusedLabelColor != other.focusedLabelColor) return false
+        if (unfocusedLabelColor != other.unfocusedLabelColor) return false
+        if (disabledLabelColor != other.disabledLabelColor) return false
+        if (errorLabelColor != other.errorLabelColor) return false
+        if (placeholderColor != other.placeholderColor) return false
+        if (disabledPlaceholderColor != other.disabledPlaceholderColor) return false
+
+        return true
+    }
+
+    override fun hashCode(): Int {
+        var result = textColor.hashCode()
+        result = 31 * result + disabledTextColor.hashCode()
+        result = 31 * result + cursorColor.hashCode()
+        result = 31 * result + errorCursorColor.hashCode()
+        result = 31 * result + focusedIndicatorColor.hashCode()
+        result = 31 * result + unfocusedIndicatorColor.hashCode()
+        result = 31 * result + errorIndicatorColor.hashCode()
+        result = 31 * result + disabledIndicatorColor.hashCode()
+        result = 31 * result + leadingIconColor.hashCode()
+        result = 31 * result + disabledLeadingIconColor.hashCode()
+        result = 31 * result + errorLeadingIconColor.hashCode()
+        result = 31 * result + trailingIconColor.hashCode()
+        result = 31 * result + disabledTrailingIconColor.hashCode()
+        result = 31 * result + errorTrailingIconColor.hashCode()
+        result = 31 * result + backgroundColor.hashCode()
+        result = 31 * result + focusedLabelColor.hashCode()
+        result = 31 * result + unfocusedLabelColor.hashCode()
+        result = 31 * result + disabledLabelColor.hashCode()
+        result = 31 * result + errorLabelColor.hashCode()
+        result = 31 * result + placeholderColor.hashCode()
+        result = 31 * result + disabledPlaceholderColor.hashCode()
+        return result
+    }
+}
+
+private const val AnimationDuration = 150
+
+//@Preview
```

Comment:
What's the matter with committing previews?

Reply:
https://github.com/AndroidDev-social/MastodonCompose/pull/67#discussion_r1024656705